src/grammar/case.rs

```diff
@@ -0,0 +1,119 @@
+use crate::{
+    grammar::parse_expr,
+    parser::{safe_loop, Parser},
+};
+use source_gen::{lexer::TokenKind, syntax::SyntaxKind, T};
+
+#[allow(unused)]
```

Comment:
Please make sure this can be used within SELECT statements, this might even be have to be put into `parse_expr`